chromium_src/components/sync_device_info/device_info_sync_bridge.cc

```diff
@@ -3,12 +3,42 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at http://mozilla.org/MPL/2.0/.
 
+#include "brave/components/sync_device_info/brave_device_info.h"
+
+#define BRAVE_MAKE_LOCAL_DEVICE_SPECIFICS \
+  specifics->mutable_brave_fields()->set_is_self_delete_supported(true);
+
 #include "../../../../components/sync_device_info/device_info_sync_bridge.cc"
 
+#undef BRAVE_MAKE_LOCAL_DEVICE_SPECIFICS
+
 #include "base/threading/sequenced_task_runner_handle.h"
 
 namespace syncer {
 
+namespace {
+
+std::unique_ptr<BraveDeviceInfo> BraveSpecificsToModel(
```

Comment:
When comparing to the total patches needed  and duplicated code for this path, I think we should just do one line patch for `SpecificsToModel`
```cpp
#define BRAVE_SPECIFICS_TO_MODEL \      
  specifics.has_brave_fields() &&
          specifics.brave_fields().has_is_self_delete_supported() && \
          specifics.brave_fields().is_self_delete_supported());
```

Reply:
Fixed at https://github.com/brave/brave-core/pull/6732/commits/db1cd081710b15a6a5ed3990a2965dfaba7ae13f , but had to  add two patched lines .

Reply:
Why do we need `BraveDeviceInfo` if we patch this way?
It is supposed to add extra field without introducing `BraveDeviceInfo` with one line patch

Reply:
ah I see, without it we can't access to the new field. 
Sorry let's revert https://github.com/brave/brave-core/commit/db1cd081710b15a6a5ed3990a2965dfaba7ae13f

Reply:
Missed your question, yes you are right. 
No worries, we hate patches :+1:  :)

Reply:
reverted